integration-templates/checkr-partner/actions/background-check-services.ts

```diff
@@ -0,0 +1,52 @@
+import type { NangoAction, CheckrServicesResponse, CheckrService } from '../../models';
+import { constructRequestWithConnectionConfig } from '../helpers/construct-request.js';
+
+/**
+ * Background Check Services
+ * @desc this handles 3 scenarios:
+ * 1. If accountHierarchyEnabled is true, then we need to fetch the services from /v1/nodes?include=packages and return the packages under their respective nodes
+ * with the associated packages
+ * 2. If accountHierarchyEnabled is false, then we need to fetch the services from /v1/packages
+ * 3. If accountHierarchyEnabled is true and there are no services under the node, then we need to fetch the services from /v1/packages
+ *
+ * This means we first fetch all the available packages. If accountHierarchyEnabled is true
+ * and there are nodes with packages we need to grab them from the packages to
+ * attach them to the node
+ */
```

Comment:
imo comments are easier to understand at the exact position, I need to do a lot of back and forth to understand why it's done this way